components/mpas-ocean/src/shared/mpas_ocn_gm.F

```diff
@@ -351,6 +351,11 @@ subroutine ocn_GM_compute_Bolus_velocity(statePool, &
                   slopeTaperDown = 1.0_RKIND + slopeTaperFactor*(slopeTaperDown - 1.0_RKIND)
 
                   sfcTaper = min(RediKappaSfcTaper(k, cell1), RediKappaSfcTaper(k, cell2))
+                  if (k < min( indMLD(cell1), indMLD(cell2))) then
+                    sfcTaper = 0.0_RKIND
+                  else
+                    sfcTaper = 1.0_RKIND
+                  end if
```

Comment:
I don't understand why the first line here isn't replaced by the last 5.  It seems like the work done in the first line now does nothing.

Reply:
Also, could something be added to explain this change?

Reply:
In particular, I'm wanting to make sure these still relate to the topic of this PR, the double counting of the Redi surface taper.

Reply:
Your last point is a very good one.  This is definitely beyond the title of the PR.  This is now more along the lines of 'generic redi fixes'.  I'll add a bit of explanation here about this change -- it is physically motivated and I believe fixes the stability issues @jonbob was seeing.  Within the upper ocean the mesoscale eddies act horizontally and transition to along isopycnal in the interior (e.g. Ferrari et al 2005), this is a very rudimentary way to achieve that.  The reason I think this will fix the issue with stability is that when I removed the surface taper double counting the redi cross terms became stronger, leading to numerical instability.  This change is effectively a much stronger taper on those terms.  I included this in my long 800 year spin up and didn't see stability issues, but Jon sees them quickly.  I can add more explanation to this PR and/or in the code.

Finally, you are absolutely right that the first line of five is not necessary.

Reply:
Thanks!  If you update the PR description, I am happy with this being included.

Reply:
more than happy to add this to the description, but I may hold off until @jonbob can test the impact to see if this stabilizes the run.